test: add comments and unittest for integral.h

pxlxingliang opened the pull request:
range: source/module_base

pxlxingliang committed:
test: add comments and unittest for integral.h
range: source/module_base

caic99 commented on source/module_base/math_integral.h:
```diff
@@ -13,7 +13,14 @@ class Integral
 
-	// Peize Lin accelerate 2017-10-02
+    /**
+     * @brief simpson integral. Peize Lin accelerate 2017-10-02
```
Author and date should be placed in a dedicated field.

pxlxingliang replied:
Where is the dedicated field. How about behind the function name?
![image](https://user-images.githubusercontent.com/91927439/151119554-9c846f4d-c88d-403e-b9ec-22f939e1e81c.png)

caic99 replied:
> Where is the dedicated field. How about behind the function name? ![image](https://user-images.githubusercontent.com/91927439/151119554-9c846f4d-c88d-403e-b9ec-22f939e1e81c.png)

"author" and "date". Read more at [doxygen manual](https://www.doxygen.nl/manual/commands.html).

That's OK!

caic99 replied:
@pxlxingliang And, I suggest you paste code in reply, but not screenshot.

caic99 commented on source/module_base/test/math_integral_test.cpp:
```diff
@@ -0,0 +1,93 @@
+    }
+};
+
+TEST_F(Simpson_Integral_sinx,Simpson_Integral_rab)
```
Both arguments TestFixtureName and TestName must be valid C++ identifiers and must not contain underscores (_). See https://google.github.io/googletest/reference/testing.html#TEST_F . 
@pxlxingliang Your former PRs also have the same problem. Sorry for my ignorance, it's my fault. Would you help me fixing them in another PR and renew test document? I suggest upper camel case is better (e.g. "SimpsonIntegralSinx").

pxlxingliang replied:
ok, I will modify it. But I don't know the reason of not containing underscores, and It seems has no any impact on the previous using of unit test.

pxlxingliang replied:
gtest already has the explanation for this.
http://google.github.io/googletest/faq.html#why-should-test-suite-names-and-test-names-not-contain-underscore

caic99 replied:
> ok, I will modify it. But I don't know the reason of not containing underscores, and It seems has no any impact on the previous using of unit test.

It's just conforms to C++ standard, and theoretically there would be no impact.

caic99 commented on source/module_base/test/math_integral_test.cpp:
```diff
@@ -0,0 +1,93 @@
+
+#include<math.h>
+
+#define PI 3.14159265358979
```
Macro `M_PI` is included in `math.h`.

pxlxingliang committed:
modify the name by removing "_" in macro TEST/TEST_F
modify the comment by moving name and date behind the function name
replace self-difined PI by the macro M_PI defined in math.in

pxlxingliang committed:
modify the comments according to the Doxygen standard

pxlxingliang commented:
deleted by mistake. reopen it

caic99 reviewed (changes requested):
Please follow up changes in #676 .

pxlxingliang committed:
Merge branch 'develop' of https://github.com.cnpmjs.org/deepmodeling/abacus-develop into integral

pxlxingliang committed:
test: modify CMakeLists.txt
